scripts/generateReleaseNotes.js

```diff
@@ -52,7 +52,9 @@ async function getReleases() {
     const branch = await run(`git rev-parse --abbrev-ref HEAD`);
     await run(`git pull origin ${branch} --tags`);
     const tags = await run(`git tag --merged ${branch} --sort=-committerdate`);
-    from = tags.split('\n')[0];
+    const list = tags.split('\n');
+    // Find the most recent tag before the current one
+    from = list.find((t, idx) => idx === list.indexOf(`v${version}`) - 1);
```

Comment:
If you want to find the tag  **before** (time-wise) then  don't you want to use a `+1`? It seems git tag orders  the tags in a descending manner, which means tags with newer dates come first.

Reply:
In this script we run `git tag --merged master --sort=-committerdate` and it uses ascending order:

```
v20.0.0-alpha1
v20.0.0-alpha2
v20.0.0-alpha3
```

Reply:
From https://git-scm.com/docs/git-tag:

> --sort=<key>
> 
>    Sort based on the key given. Prefix - to sort in descending order of the value

As we are using the `-` prefix it should be descending. Remove the minus to get ascending order. I can also verify this behaviour on a local repo.

Reply:
It seems like the script uses incorrect format, here is the fixed version to use descending order:

```
git tag --merged master --sort='-*committerdate'
```

Could you try if that works please? If that looks good, I will update the script accordingly.

Reply:
Mhh, that still returns them in descending order for me:
```
(linkding) ⋊> ~/d/p/linkdings on master ⨯ git tag --merged master --sort='-*committerdate'                                                                                                                                                                                                                                                                                                                                    
v1.6.2
v1.6.1
v1.6.0
v1.5.0
v1.4.1
v1.4.0
```
Anyway descending or ascending is not really the issue, you just want to pick the correct tag which you can do by picking either the tag above or below whatever is referenced by `version` - here I'm not sure I understand the logic completely, I was just picking on the fact that the comment stated `before` but then was using something `later`. If you really want the tag before the current one (time-wise), and the tags are ordered descending (time-wise) then use a `+1` 🙂

Reply:
Thanks. For me the original script returned the tags in ascending order, that's why I used `-1`.

Reply:
Updated to simplify the script using the command above.